### ISSUE TITLE
Accordion and Accordion Item - Ensure custom controls provide proper textual name, role, and state information - (2036569668)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-design-system#5553]
### Module: 
 03 Accordion and Accordion Item 

### Violation: 
 Ensure custom controls provide proper textual name, role, and state information 

![image](https://user-images.githubusercontent.com/5023024/202766183-52688513-9ed0-47d6-9af1-d7c4045e4224.png)

### WCAG Reference: 
- [1.1.1 Non-text Content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html)
- [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) 

### Instance ID: 
 2036569668 

### Severity: 
 7 

### Description: 
 [Issue]
Accordions lack full implementation, missing some roles and/or states information. 

[User Impact]
Screen reader users will be unable to determine that these controls expand and collapse content. They will also have difficulty determining whether the controls are currently expanded or collapsed.

[Code Reference] 
 ```
<calcite-accordion appearance="default" icon-position="end" icon-type="chevron" scale="m" selection-mode="single" calcite-hydrated=""> ... </calcite-accordion>
 ``` 

### Note: 
 [Suggestion]
Ensure custom controls provide proper textual name, role, and state information.

For accordions, the following information is expected:
- Each accordion header must be enclosed in a ``` <button> ``` element.
- Each of these buttons must be enclosed in a heading element of an appropriate level. The ``` <button> ``` element must be the only child of this heading element.
- When the panel is visible, the header button must have aria-expanded="true". When it is hidden, the header button must have aria-expanded="false".
- The button must have aria-controls set to the ID of the panel container.
- If a panel is visible and locked in the expanded state, its header button must have aria-disabled="true".

In addition, it is recommended that container elements have a region role as it greatly increases accessibility for non-visual users.  This is not required though.  More information in the link below.

[Recommended Reading]
For more information about creating accessible accordions, please see the W3C Web Accessibility Initiative&#039;s ARIA Authoring Practices Guide: https://www.w3.org/WAI/ARIA/apg/patterns/accordion/ 

### Media Type: 
 Custom Controls 

### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by brittneytewks** [comment]
Removed "needs triage" since a priority is set and estimate is indicated

**Comment by geospatialem** [comment]
Similar to https://codepen.io/geospatialem/pen/gOBveyL, to ensure AT users are able to receive context upon expansion/collapsing of `accordion-item`, we should add the following to the component:
- to the [`div` above the `slot`](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/accordion-item/accordion-item.tsx#L217)
    - `aria-live="polite"`
    - `role="region"` and
    - unique `id` which can be linked in the  [`div` with a `role="button"`](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/accordion-item/accordion-item.tsx#L186-L192) (_below_)
- to the [`div` with a `role="button"`](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/accordion-item/accordion-item.tsx#L186-L192)
    - an `aria-labelledby` using the `id` from the `div` above the slot (_above_)

**Comment by geospatialem** [comment]
Verified on the `main` branch.
